dhcpv6/option_vendor_opts.go

```diff
@@ -0,0 +1,117 @@
+package dhcpv6
+
+/*
+ This module defines the OptVendorOpts structure.
+ https://tools.ietf.org/html/rfc3315#section-22.17
+
+				Option 17
+       0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |      OPTION_VENDOR_OPTS       |           option-len          |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |                       enterprise-number                       |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      .                                                               .
+      .                   option-data (sub-options)                   .
+      .                                                               .
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+				Sub-Option
+	 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	|          opt-code             |             option-len        |
+	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	.                                                               .
+	.                          option-data                          .
+	.                                                               .
+	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+*/
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+)
+
+// OptVendorOpts represents a DHCPv6 Status Code option
+type OptVendorOpts struct {
+	EnterpriseNumber uint32
+	VendorOpts       []Option
+}
+
+// Code returns the option code
+func (op *OptVendorOpts) Code() OptionCode {
+	return OptionVendorOpts
+}
+
+// ToBytes serializes the option and returns it as a sequence of bytes
+func (op *OptVendorOpts) ToBytes() []byte {
+	buf := make([]byte, 8)
+	binary.BigEndian.PutUint16(buf[0:2], uint16(OptionVendorOpts))
+	binary.BigEndian.PutUint16(buf[2:4], uint16(op.Length()))
+	binary.BigEndian.PutUint32(buf[4:8], uint32(op.EnterpriseNumber))
+	for _, opt := range op.VendorOpts {
+		buf = append(buf, opt.ToBytes()...)
+	}
+	return buf
+}
+
+// Length returns the option length
+func (op *OptVendorOpts) Length() int {
+	l := 4 // 4 bytes for Enterprise Number
+	for _, opt := range op.VendorOpts {
+		l += 4 + opt.Length() // 4 bytes for Code and Length from Vendor
```

Comment:
the comment should be on a line on its own as per Go formatting rules. I'm surprised that Stickler didn't catch it!